### ISSUE TITLE
How to design a binary to access it on startup

### ISSUE THREAD

**Issue by gedw99** [zalando/go-keyring#57]
I have a configuration file that holds a password that my app uses to encrypt my DB.

That configuration file is encrypted against a password. 

How can i make it so that only that app can access the password i stored into the TPM / KeyRing ?

The intent here is that is someone gets physical access to the machines hard drive they cannot get the configuration data stored in the config file.

Currently i store the password to decrypt the configuration file in Systemd file.

**Comment by mikkeloscar** [comment]
I don't think this is something this project is useful for. It's based on the system keyrings where the use case is the user of the machine knows the keyring password and thereby can unlock whatever is stored there. If you can't provide the keyring password temporarily from "outside" then this doesn't protect anything.

**Comment by merusso** [comment]
Storing a symmetric encryption key in the keyring should be more secure than storing it in a file. It would be better secured against physical access as the keyring's contents are encrypted at rest.

The service and user values aren't meant to be sensitive. They can be set as plaintext config values passed to your app. The password is sensitive and should only be stored in the keyring.
